Hitting `-Wvla` warnings-as-errors on the SaferCPP bot:

```
/Volumes/Data/worker/macOS-Safer-CPP-Checks-EWS/build/Source/JavaScriptCore/API/tests/testapi.c:1198:24: error: variable
 length array used [-Werror,-Wvla]
 1198 |     char sourceCString[sourceCStringSize];
      |                        ^~~~~~~~~~~~~~~~~
/Volumes/Data/worker/macOS-Safer-CPP-Checks-EWS/build/Source/JavaScriptCore/API/tests/testapi.c:1203:21: error: variable
 length array used [-Werror,-Wvla]
 1203 |     char outCString[outCStringSize];
      |                     ^~~~~~~~~~~~~~
/Volumes/Data/worker/macOS-Safer-CPP-Checks-EWS/build/Source/JavaScriptCore/API/tests/testapi.c:1229:24: error: variable
 length array used [-Werror,-Wvla]
 1229 |     char sourceCString[sourceCStringSize];
      |                        ^~~~~~~~~~~~~~~~~
/Volumes/Data/worker/macOS-Safer-CPP-Checks-EWS/build/Source/JavaScriptCore/API/tests/testapi.c:1239:21: error: variable
 length array used [-Werror,-Wvla]
 1239 |     char outCString[outCStringSize];
      |                     ^~~~~~~~~~~~~~
/Volumes/Data/worker/macOS-Safer-CPP-Checks-EWS/build/Source/JavaScriptCore/API/tests/testapi.c:1265:24: error: variable
 length array used [-Werror,-Wvla]
 1265 |     char sourceCString[sourceCStringSize];
      |                        ^~~~~~~~~~~~~~~~~
/Volumes/Data/worker/macOS-Safer-CPP-Checks-EWS/build/Source/JavaScriptCore/API/tests/testapi.c:1275:21: error: variable length array used [-Werror,-Wvla]
 1275 |     char outCString[outCStringSize];
      |                     ^~~~~~~~~~~~~~
6 errors generated.
```

--- a/Source/JavaScriptCore/API/tests/testapi.c
+++ b/Source/JavaScriptCore/API/tests/testapi.c
@@ -1192,42 +1192,48 @@ void JSSynchronousGarbageCollectForDebugging(JSContextRef);
 static void checkJSStringOOBUTF8(void)
 {
     const size_t sourceCStringSize = 200;
-    const size_t outCStringSize = 10;
+    const size_t cStringSize = 10;
+    const size_t outCStringSize = cStringSize + sourceCStringSize;
 
-    char* sourceCString = (char*)malloc(sourceCStringSize);
-    memset(sourceCString, 0, sourceCStringSize);
+IGNORE_WARNINGS_BEGIN("vla")
+    char sourceCString[sourceCStringSize];
+IGNORE_WARNINGS_END
+    memset(sourceCString, 0, sizeof(sourceCString));
     for (size_t i = 0; i < sourceCStringSize - 1; ++i)
         sourceCString[i] = '0' + (i%10);
 
-    char* outCString = (char*)malloc(outCStringSize + sourceCStringSize);
-    memset(outCString, 0x13, outCStringSize + sourceCStringSize);
+IGNORE_WARNINGS_BEGIN("vla")
+    char outCString[outCStringSize];
+IGNORE_WARNINGS_END
+    memset(outCString, 0x13, sizeof(outCString));
 
     JSStringRef str = JSStringCreateWithUTF8CString(sourceCString);
-    size_t bytesWritten = JSStringGetUTF8CString(str, outCString, outCStringSize);
+    size_t bytesWritten = JSStringGetUTF8CString(str, outCString, cStringSize);
 
     assertTrue(bytesWritten == 10, "we report 10 bytes written precisely");
 
     for (size_t i = 0; i < sizeof(outCString); ++i) {
-        if (i == outCStringSize - 1)
+        if (i == cStringSize - 1)
             assertTrue(outCString[i] == '\0', "string terminated");
-        else if (i < outCStringSize - 1)
+        else if (i < cStringSize - 1)
             assertTrue(outCString[i] == sourceCString[i], "string copied");
         else
             assertTrue(outCString[i] == 0x13, "did not write past the end");
     }
 
     JSStringRelease(str);
-    free(outCString);
-    free(sourceCString);
 }
 
 static void checkJSStringOOBUTF16(void)
 {
     const size_t sourceCStringSize = 22;
-    const size_t outCStringSize = 20;
+    const size_t cStringSize = 20;
+    const size_t outCStringSize = cStringSize + sourceCStringSize;
 
-    char* sourceCString = (char*)malloc(sourceCStringSize);
-    memset(sourceCString, 0, sourceCStringSize);
+IGNORE_WARNINGS_BEGIN("vla")
+    char sourceCString[sourceCStringSize];
+IGNORE_WARNINGS_END
+    memset(sourceCString, 0, sizeof(sourceCString));
     for (size_t i = 0; i < sourceCStringSize - 1; ++i)
         sourceCString[i] = '0' + (i%10);
 
@@ -1236,35 +1242,38 @@ static void checkJSStringOOBUTF16(void)
     sourceCString[5] = '\x98';
     sourceCString[6] = '\x81';
 
-    char* outCString = (char*)malloc(outCStringSize + sourceCStringSize);
-    memset(outCString, 0x13, outCStringSize + sourceCStringSize);
+IGNORE_WARNINGS_BEGIN("vla")
+    char outCString[outCStringSize];
+IGNORE_WARNINGS_END
+    memset(outCString, 0x13, sizeof(outCString));
 
     JSStringRef str = JSStringCreateWithUTF8CString(sourceCString);
-    size_t bytesWritten = JSStringGetUTF8CString(str, outCString, outCStringSize);
+    size_t bytesWritten = JSStringGetUTF8CString(str, outCString, cStringSize);
 
     assertTrue(bytesWritten == 20, "we report 20 bytes written precisely");
 
     for (size_t i = 0; i < sizeof(outCString); ++i) {
-        if (i == outCStringSize - 1)
+        if (i == cStringSize - 1)
             assertTrue(outCString[i] == '\0', "string terminated");
-        else if (i < outCStringSize - 1)
+        else if (i < cStringSize - 1)
             assertTrue(outCString[i] == sourceCString[i], "string copied");
         else
             assertTrue(outCString[i] == 0x13, "did not write past the end");
     }
 
     JSStringRelease(str);
-    free(outCString);
-    free(sourceCString);
 }
 
 static void checkJSStringOOBUTF16AtEnd(void)
 {
     const size_t sourceCStringSize = 22;
-    const size_t outCStringSize = 20;
+    const size_t cStringSize = 20;
+    const size_t outCStringSize = cStringSize + sourceCStringSize;
 
-    char* sourceCString = (char*)malloc(sourceCStringSize);
-    memset(sourceCString, 0, sourceCStringSize);
+IGNORE_WARNINGS_BEGIN("vla")
+    char sourceCString[sourceCStringSize];
+IGNORE_WARNINGS_END
+    memset(sourceCString, 0, sizeof(sourceCString));
     for (size_t i = 0; i < sourceCStringSize - 1; ++i)
         sourceCString[i] = '0' + (i%10);
 
@@ -1273,11 +1282,13 @@ static void checkJSStringOOBUTF16AtEnd(void)
     sourceCString[19] = '\x98';
     sourceCString[20] = '\x81';
 
-    char* outCString = (char*)malloc(outCStringSize + sourceCStringSize);
-    memset(outCString, 0x13, outCStringSize + sourceCStringSize);
+IGNORE_WARNINGS_BEGIN("vla")
+    char outCString[outCStringSize];
+IGNORE_WARNINGS_END
+    memset(outCString, 0x13, sizeof(outCString));
 
     JSStringRef str = JSStringCreateWithUTF8CString(sourceCString);
-    size_t bytesWritten = JSStringGetUTF8CString(str, outCString, outCStringSize);
+    size_t bytesWritten = JSStringGetUTF8CString(str, outCString, cStringSize);
 
     assertTrue(bytesWritten == 18, "we report 18 bytes written precisely");
 
@@ -1291,8 +1302,6 @@ static void checkJSStringOOBUTF16AtEnd(void)
     }
 
     JSStringRelease(str);
-    free(outCString);
-    free(sourceCString);
 }
 
 static void checkJSStringOOB(void)

